### ISSUE TITLE
Expose underlining information for external integrations

### ISSUE THREAD

**Issue by archSeer** [rrrene/credo#471]
I'm using the [ale](https://github.com/w0rp/ale) plugin for vim to automatically run and provide inline checks inside my code editor. ale supports underlining specific parts of the line to show where exactly the error is. Here is an example of the rubocop integration:

<img width="717" alt="screen shot 2017-11-13 at 15 27 38" src="https://user-images.githubusercontent.com/1372918/32712317-d20f8d30-c83b-11e7-9220-204edda45347.png">

In order for that to work, it needs a line number `lnum`, `col` and `end_col`. The current integration [uses `--format=flycheck` to retrieve the line number and column, but has no way of retrieving the `end_col`](https://github.com/w0rp/ale/blob/584e0bc7f25563bf4ab3ae738b78d9d13a898f94/ale_linters/elixir/credo.vim). Contrast that with the [rubocop linter](https://github.com/w0rp/ale/blob/master/ale_linters/ruby/rubocop.vim#L36) which returns json (col + length).

Since credo already underlines warnings on `--format=oneline`, Would it be possible to expose that information somehow? Maybe expose a new `--format=json`, similar to rubocop?

**Comment by rrrene** [comment]
@archSeer Sounds like a good idea :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now part of `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

and run Credo with

```shell
mix credo --format=json
```

Please report back if your issue is solved! 👍

**Comment by archSeer** [comment]
Works great! Made a patch for ALE, I'll submit it once a new credo release is pushed:

<img width="691" alt="screen shot 2018-01-22 at 20 11 18" src="https://user-images.githubusercontent.com/1372918/35218001-02c5a6c0-ff65-11e7-9a99-42b9d8f0b27a.png">
